qiime2/sdk/usage.py

```diff
@@ -284,30 +691,50 @@ def action(self, action: UsageAction, inputs: UsageInputs,
         computed_outputs = self._action_(action, input_opts, output_opts)
         self._add_outputs_to_scope(outputs, computed_outputs)
 
-    def _action_(self, action: UsageAction,
-                 input_opts: dict, output_opts: dict):
+    def _action_(
+        self, action: UsageAction, input_opts: dict, output_opts: dict
+    ) -> dict:
         raise NotImplementedError
 
     def _assert_has_line_matching_(self, ref, label, path, expression):
         raise NotImplementedError
 
-    def get_result(self, ref):
+    def get_result(self, ref: str) -> ScopeRecord:
+        """
+        Get the record for a Usage example output. This is a convenience
+        method used to access records generated after running `Usage.action`.
+
+        Parameters
+        ----------
+        ref : str
+            Output name
+
+        Raises
+        ------
+            If `ref` is not associated with a record generated by
+            `Usage.action`
+        """
         record = self._get_record(ref)
         source = record.source
         if source != 'action':
             raise TypeError('source == %s but must be "action"' % source)
         return record
 
-    def _add_outputs_to_scope(self, outputs, computed_outputs):
+    def _add_outputs_to_scope(
+        self, outputs: UsageOutputNames, computed_outputs
+    ):
         outputs.validate_computed(computed_outputs)
-        for output, result in computed_outputs.items():
+        for output, value in computed_outputs.items():
```

Comment:
```suggestion
        for output, result in computed_outputs.items():
```

`Result` is the base-class for `Artifact` and `Visualization`, so `result` here is deliberate, to indicate that this is some kind of `Result` object.